File: training/avro/src/main/java/br/com/devcave/avro/service/Producer.java
package br.com.devcave.avro.service;

import br.com.devcave.avro.domain.User;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.kafka.core.KafkaTemplate;
import org.springframework.stereotype.Component;

@Slf4j
@Component
@RequiredArgsConstructor
public class Producer {

    private final KafkaTemplate<String, User> kafkaTemplate;

    public void sendMessage(final User user) {
        this.kafkaTemplate.send("user-replication", user.getFirstName(), user);
        log.info(String.format("Produced user -> %s", user));
    }
}

File: kafka-avro/spring-kafka-avro/src/main/kotlin/br/com/devcave/avro/service/Producer.kt
package br.com.devcave.avro.service

import br.com.devcave.avro.domain.User
import org.slf4j.LoggerFactory
import org.springframework.kafka.core.KafkaTemplate
import org.springframework.stereotype.Component
import kotlin.random.Random

@Component
class Producer(
    private val kafkaTemplate: KafkaTemplate<String, User>
) {
    private val logger = LoggerFactory.getLogger(javaClass)

    fun send() {
        val user = User.newBuilder()
            .setFirstName("Uri")
            .setLastName("Cana")
            .setAge(41)
            .setHeight(170f)
            .build()
        logger.info("send, user={}", user)
        kafkaTemplate.send("user-avro", Random.nextInt().toString(), user)
    }
}